homeassistant/components/soundtouch/media_player.py

```diff
@@ -350,6 +352,17 @@ def play_media(self, media_type, media_id, **kwargs):
             else:
                 _LOGGER.warning("Unable to find preset with id %s", media_id)
 
+    def select_source(self, source):
+        """Select input source."""
+        if source == "AUX":
+            _LOGGER.debug("Selecting source AUX")
+            self._device.select_source_aux()
+        elif source == "Bluetooth":
+            _LOGGER.debug("Selecting source Bluetooth")
+            self._device.select_source_bluetooth()
+        else:
+            _LOGGER.warning("Source %s is not supported", source)
+
```

Comment:
You should also implement the `source` and `source_list` properties:
https://github.com/home-assistant/home-assistant/blob/41f3fb291afe6596f276400e8c692e815172f40d/homeassistant/components/media_player/__init__.py#L483-L491

Reply:
I'll try to do that, but I'm not very confident. We'll see